File: lib/groq.ts
// lib/groq.ts
import { groq } from '@ai-sdk/groq';

export async function interpretCommand(userInput: string) {
  const response = await groq.chat.completions.create({
    messages: [{ role: 'user', content: userInput }],
    model: 'mixtral-8x7b-32768',
  });

  const raw = response.choices[0].message.content;
  return parseGroqResponse(raw);
}

function parseGroqResponse(content: string) {
  const lower = content.toLowerCase();

  if (lower.includes("summarize") && lower.includes("email"))
    return { action: "summarize_emails", targetApp: "Outlook" };

  if (lower.includes("crm") && lower.includes("interaction"))
    return { action: "fetch_crm_interactions", targetApp: "CRM" };

  if (lower.includes("calendar") && lower.includes("event"))
    return { action: "fetch_calendar_events", targetApp: "Calendar" };

  if (lower.includes("open") && lower.includes("notepad"))
    return { action: "open_app", targetApp: "Notepad" };

  return { action: "unknown", raw: content };
}

File: lib/groq.py
def process_groq_command(command: str) -> dict:
    """
    Process the user command and return the action and parameters.
    :param command: User command as a string.
    :return: Dictionary with action and parameters.
    """
    command = command.lower()

    if "summarize emails" in command:
        return {"action": "summarize_emails", "parameters": {"email_count": 5, "folder": "inbox"}}
    elif "fetch crm interactions" in command:
        return {"action": "fetch_crm_interactions", "parameters": {}}
    elif "fetch calendar events" in command:
        return {"action": "fetch_calendar_events", "parameters": {}}
    elif "open notepad" in command:
        return {"action": "open_app", "parameters": {"app_path": "notepad.exe"}}
    elif "type hello world" in command:
        return {"action": "type_text", "parameters": {"text": "Hello, World!"}}
    else:
        return {"action": "unknown", "parameters": {"raw_command": command}}